src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchMultiSourceETLTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.etl;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import com.sforce.soap.metadata.CustomField;
+import com.sforce.soap.partner.DescribeGlobalSObjectResult;
+import com.sforce.soap.partner.sobject.SObject;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.plugin.salesforce.plugin.source.batch.multi.SalesforceMultiSourceConfig;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * {@inheritDoc}
+ */
+public class SalesforceBatchMultiSourceETLTest extends BaseSalesforceBatchSourceETLTest {
+
+  @Test
+  public void testWhiteList() throws Exception {
+    String sObjectName1 = createCustomObject("IT_Multi_WL_1",
+      new CustomField[] {createTextCustomField("Position__c")});
+    String sObjectName2 = createCustomObject("IT_Multi_WL_2",
+      new CustomField[] {createTextCustomField("City__c")});
+
+    List<SObject> sObjects = ImmutableList.of(
+      new SObjectBuilder()
+        .setType(sObjectName1)
+        .put("Name", "Fred")
+        .put("Position__c", "DEV")
+        .build(),
+      new SObjectBuilder()
+        .setType(sObjectName1)
+        .put("Name", "Wilma")
+        .put("Position__c", "QA")
+        .build(),
+      new SObjectBuilder()
+        .setType(sObjectName2)
+        .put("Name", "Pebbles")
+        .put("City__c", "NY")
+        .build());
+
+    addSObjects(sObjects, false);
+
+    List<StructuredRecord> records = getResultsForMultiSObjects(
+      String.join(",", sObjectName1, sObjectName2), null);
+
+    String sObjectNameField = SalesforceMultiSourceConfig.SOBJECT_NAME_FIELD_DEFAULT;
+
+    Set<String> expectedResults = Sets.newHashSet(
```

Comment:
Instead of converting the record into a string, can we build what the expected records should look like for each, and then compare the records?

Reply:
If the values are just strings, you be able to build the records like Jay mentioned and Assert.assertEquals(expectedRecords, actualRecords).

It's only if the records contain bytes, arrays, and other things that don't compare nicely in Java that you need to explicitly compare field values.

Reply:
Done.